tests/telemetry/test_telemetry.py

```diff
@@ -222,14 +229,6 @@ def test_on_change_updates(duthosts, enum_rand_one_per_hwsku_hostname, ptfhost,
 
     client_thread.join(30)
```

Comment:
>30 [](http://example.com/codeflow?start=23&length=2)

This is old code. Sleeping for magic number seconds is problematic. Even you already finished the core test earlier, you still need to waste time?
Suggest early terminate, or wait on event.

Reply:
The 30 seconds is a max timeout for how long gnmi client inside ptf docker will run for before we terminate it. If we get the update that we want, gnmi client will close with message "Max update reached", and the thread will join before the max timeout of 30 seconds. However, if we don't get event, we will give a max amount of 30 seconds before we time the client out and close the thread.